source/fuzz/fuzzer_util.cpp

```diff
@@ -1362,18 +1362,27 @@ opt::Instruction* GetLastInsertBeforeInstruction(opt::IRContext* ir_context,
 }
 
 bool HasBuiltInMembers(opt::IRContext* ir_context, uint32_t struct_type_id) {
-  const auto* type = ir_context->get_type_mgr()->GetType(struct_type_id);
-  (void)type;  // Make compilers happy in release mode.
-  assert(type && type->AsStruct() &&
+  const auto* type_inst = ir_context->get_def_use_mgr()->GetDef(struct_type_id);
+  assert(type_inst && type_inst->opcode() == SpvOpTypeStruct &&
          "|struct_type_id| is not a result id of an OpTypeStruct");
 
-  return !ir_context->get_def_use_mgr()->WhileEachUser(
-      struct_type_id, [struct_type_id](const opt::Instruction* user) {
-        return user->opcode() != SpvOpMemberDecorate ||
-               user->GetSingleWordInOperand(0) != struct_type_id ||
-               static_cast<SpvDecoration>(user->GetSingleWordInOperand(2)) !=
-                   SpvDecorationBuiltIn;
+  uint32_t builtin_count = 0;
+  ir_context->get_def_use_mgr()->ForEachUser(
+      type_inst,
+      [struct_type_id, &builtin_count](const opt::Instruction* user) {
+        if (user->opcode() == SpvOpMemberDecorate &&
+            user->GetSingleWordInOperand(0) == struct_type_id &&
+            static_cast<SpvDecoration>(user->GetSingleWordInOperand(2)) ==
+                SpvDecorationBuiltIn) {
+          ++builtin_count;
+        }
       });
+
+  assert((builtin_count == 0 || builtin_count == type_inst->NumInOperands()) &&
+         "The module is invalid: either one or none of the members of "
```

Comment:
"either one" -> "either all", I think.  And perhaps make it "none or all" rather than "all or none" to match the order of the operands to "||" in the assertion.

Reply:
Fixed in 9868ff3.